server/region_handler.go

```diff
@@ -0,0 +1,486 @@
+	goctx "golang.org/x/net/context"
+)
+
+const (
+	pDBName    = "db"
+	pTableName = "table"
+	pRegionID  = "regionID"
+)
+
+const (
+	headerContentType = "Content-Type"
+	contentTypeJSON   = "application/json"
+)
+
+// TableRegions is the response data for list table's regions.
+// It contains regions list for record and indices.
+type TableRegions struct {
+	TableName     string         `json:"name"`
+	TableID       int64          `json:"id"`
+	RecordRegions []RegionMeta   `json:"record_regions"`
+	Indices       []IndexRegions `json:"indices"`
+}
+
+// RegionMeta contains a region's peer detail
+type RegionMeta struct {
+	ID          uint64              `json:"region_id"`
+	Leader      *metapb.Peer        `json:"leader"`
+	Peers       []*metapb.Peer      `json:"peers"`
+	RegionEpoch *metapb.RegionEpoch `json:"region_epoch"`
+}
+
+// IndexRegions is the region info for one index.
+type IndexRegions struct {
+	Name    string       `json:"name"`
+	ID      int64        `json:"id"`
+	Regions []RegionMeta `json:"regions"`
+}
+
+// RegionDetail is the response data for get region by ID
+// it includes indices and records detail in current region.
+type RegionDetail struct {
+	RegionID uint64      `json:"region_id"`
+	StartKey []byte      `json:"start_key"`
+	EndKey   []byte      `json:"end_key"`
+	Frames   []FrameItem `json:"frames"`
+}
+
+// FrameItem includes a index's or record's meta data with table's info.
+type FrameItem struct {
+	DBName    string `json:"db_name"`
+	TableName string `json:"table_name"`
+	TableID   int64  `json:"table_id"`
+	IsRecord  bool   `json:"is_record"`
+	IndexName string `json:"index_name,omitempty"`
+	IndexID   int64  `json:"index_id,omitempty"`
+}
+
+// RegionFrameRange contains a frame range info which the region covered.
+type RegionFrameRange struct {
+	first  *FrameItem        // start frame of the region
+	last   *FrameItem        // end frame of the region
+	region *tikv.KeyLocation // the region
+}
+
+// RegionHandler is the common field for http region handler. It contains
+// some common functions which would be used in TableRegionsHandler and
+// HTTPGetRegionByIDHandler.
+type RegionHandler struct {
+	pdClient pd.Client
+	server   *Server
+}
+
+// TableRegionsHandler is the handler for list table's regions.
+type TableRegionsHandler struct {
+	RegionHandler
+}
+
+// regionHandlerTool contains some common tool which
+// would be used in each RegionHandler request.
+type regionHandlerTool struct {
+	bo          *tikv.Backoffer
+	infoSchema  infoschema.InfoSchema
+	regionCache *tikv.RegionCache
+}
+
+func (s *Server) newRegionHandler(pdClient pd.Client) RegionHandler {
+	return RegionHandler{
+		pdClient: pdClient,
+		server:   s,
+	}
+}
+
+func (s *Server) newTableRegionsHandler(pdClient pd.Client) http.Handler {
+	return TableRegionsHandler{
+		s.newRegionHandler(pdClient),
+	}
+}
+
+func (rh TableRegionsHandler) getRegionsMetaWithRegionIds(rIDs []uint64) ([]RegionMeta, error) {
```

Comment:
`Id` should be `ID`. Also I think just `getRegionMetas(regionIDs []uint64)` is ok.